src/components/map/MapComponent.vue

```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+import { useLayersStore } from '@/stores/layers'
+import type { MapCollection } from '@vcmap/core'
+import { onMounted, ref } from 'vue'
+import type { Ref } from 'vue'
+import { prepareContext } from '../../services/vcmap/context'
+import UiButton from '../ui/UiButton.vue'
+import UiMap from '../ui/UiMap.vue'
+import TransportButtons from './TransportButtons.vue'
+
+let mapCollection: Ref<MapCollection | undefined> = ref(undefined)
+const layerStore = useLayersStore()
+
+onMounted(async () => {
+  mapCollection.value = await prepareContext()
+})
+
+function setLayerVisible(layerName: string, visible: boolean) {
+  const layer = mapCollection.value?.layerCollection.getByKey(layerName)
+  if (visible) {
+    layer?.activate()
+  } else if (layer?.active) {
+    layer.deactivate()
+  }
+}
+
+function is3D(): boolean {
+  return mapCollection.value?.activeMap.name === 'cesium'
+}
+
+function toggleMap() {
+  mapCollection.value?.setActiveMap(is3D() ? 'ol' : 'cesium')
```

Comment:
Could you tell me the difference between the active map and the one in the layer collection?

Reply:
A map collection is a set of `VcsMaps`, openlayers, cesium etc.. The active map is just the one which is active, eg. 2D or 3D.
The layer collection is the set of layers for the map collection (not for the maps), active layers are the one available in the maps. Some layers are just active in 3D mode though (terrain, buildings etc...).